apimodel/__init__.py

```diff
@@ -103,18 +101,34 @@ def __getattr__(self, field):
 
         if field in self.fields:
             type_ = self._string_to_class(self.fields[field])
-            if isinstance(type_, list):
-                collection_type = self._string_to_class(type_[0])
-                collection = []
-                for data in self._data.get(field, []):
-                    collection.append(
-                        self._get_submodels_lazily(collection_type, data))
-                result = collection
+            if isinstance(type_, APICollection) or \
+                    issubclass(type_, APICollection):
+                if field in self.collection_finders:
+                    result = type_(
+                        data=self.collection_finders[field].format(self),
+                        lazy_load=True,
+                    )
+                else:
+                    data = self._data.get(field, [])
+                    if isinstance(type_, APICollection):
+                        result = APICollection(model=type_.model, data=data,
```

Comment:
This is the sloppiest part I think... On-the-fly `APICollection`s can be created with `APICollection(model=MyModel, lazy_load=True)`. Maybe it would be better if application code wasn't supposed to do this and had to declare their collections manually.